source/default/_patterns/00-protons/index.js

```diff
@@ -1,9 +1,37 @@
 /**
- * Base css generation and global js logic
+ * Base css generation and global js logic.
  */
 
 import './_base.scss';
 
+import enquire from 'enquire.js';
+import $ from 'jquery';
+import { mediaBreakpoint } from './utilities';
+
+// Get the breakpoints set to :root by _bootstrap-overrides.scss.
+const breakpointsString = $('html').css('--breakpoints');
+const breakpoints = {};
+
+if (breakpointsString) {
+  breakpointsString.split(' ').forEach(breakpoint => {
+    const value = $('html').css(`--breakpoint-${breakpoint}`);
+    if (value) {
+      breakpoints[breakpoint] = value;
+    }
+  });
+}
```

Comment:
There's a really cool chance to use some new tools here. Annotated below.

```js
// Since we'll reference this multiple times, var it
const $root = $(':root');

// We're going to chain down this. It's guaranteed to be at least an empty object.
const breakpoints = $root
  // Read the :root style like before
  .css('--breakpoints')
  // Remove whitespace (see Sass change below)
  .replace(/ /g, '')
  // Split on comma (see Sass change below)
  .split(',')
  // Starting with an empty object (second arg to .reduce()), return a new object
  // that spreads the prior keys into it plus our new key + value
  .reduce(
    (bps, bp) => ({
      ...bps,
      [bp]: $root.css(`--breakpoint-${bp}`),
    }),
    {}
  );
```

Reply:
@illepic This is all great stuff. The one change I had to make was to pull the `.replace` out due to CI running it prior to finding `$root.css('--breakpoints')`, as best I can figure. Bottom line is I was getting "can't split on undefined". So you'll see that difference in the latest push.